pilot/pkg/serviceregistry/kube/controller/filter/discoverynamespaces.go

```diff
@@ -71,8 +71,14 @@ func (d *discoveryNamespacesFilter) Filter(obj interface{}) bool {
 		return true
 	}
 
+	// When an object is deleted, obj could be a DeletionFinalStateUnknown marker item.
```

Comment:
Can you process `DeletionFinalStateUnknown` as well, so we donot miss a delete event

Reply:
good call - done